### ISSUE TITLE
Autoplay hoc won't work 

### ISSUE THREAD

**Issue by Nykollas** [rcaferati/react-awesome-slider#86]
The autoplay hoc seems not work properly.
Just stucked on the first image

```jsx
import AwesomeSlider from  'react-awesome-slider';
import withAutoplay from 'react-awesome-slider/dist/autoplay';
const AutoPlayer = withAutoplay(AwesomeSlider);

export default AwesomeComponent extends Component{
 render = () => {
        return (
            <>
                {this.state.data ?
                    <AutoPlayer play={true}
                    cancelOnInteraction={false} // should stop playing on user interaction
                    interval={1000}>
                        {this.state.data.map((item) => {
                                    return(
                                        <div className="template-container">
                                            <div className="template-img-container">
                                                    <div className="template-img">
                                                        <img src={item.img}></img>
                                                    </div>
                                            </div>
                                            <div className="descr-container">       
                                                        <div className='descr-box'>
                                                            <p className="descr-text">{item.caption}</p>
                                                        </div>
                                            </div>
                                        </div>)
                        })}
                    </AutoPlayer>
                    : <></>
                }
            </>
        );
    }

```

**Comment by cbr9** [comment]
I'm not sure, but it may have something to do with the fact that you're wrapping the images in an extra div, and also they have a sibling.

**Comment by NisugaJ** [comment]
Having the same issue here also.

`import React, { useState, useEffect } from "react"
import AwesomeSlider from "react-awesome-slider"
import withAutoplay from "react-awesome-slider/dist/autoplay"
import "react-awesome-slider/dist/styles.css"
import { Box } from "@material-ui/core"
import baseAxios from "../config/axios"
import backend_config from "../config/backend_config"

const AutoplaySlider = withAutoplay(AwesomeSlider)

const VacancySlideShow = () => {
  const [allPhotos, setAllPhotos] = useState({ data: [] })

  async function fetchDataForAppSlide() {
    let ignore = false
    try {
      const result_CurrentPositions = await baseAxios.get("/positions")
      if (!ignore && result_CurrentPositions.data) {
        let photos = result_CurrentPositions.data.map(
          ({ imagePath }) => imagePath
        )
        setAllPhotos({ data: photos })
      }
    } catch (error) {
      console.log(error)
    }
  }

  useEffect(() => {
    fetchDataForAppSlide()
    console.log("fetched")
  }, [])

  return (
    <Box>
      <AutoplaySlider play={true} cancelOnInteraction={false} interval={6000}>
        {allPhotos.data.map(item => (
          <Box
            key={item}
            style={{ borderRadius: 5 }}
            data-src={backend_config.baseURL + item}
          />
        ))}
      </AutoplaySlider>
    </Box>
  )
}

export default VacancySlideShow`

Gives following error. 
![image](https://user-images.githubusercontent.com/44756803/73346754-6be1f500-42ac-11ea-8436-7ca713a1a88e.png)

**Comment by rcaferati** [comment]
@Nykollas @NisugaJ thanks for raising it. Can you guys try `v3.2.2` and check if it is still happening? I've done a couple fixes on the Autoplay regarding having an empty children state.

@NisugaJ Either way you could add a startupScreen component or avoid the slider to be mounted when there's no `data` on the `allPhotos` state. That could also do it.

**Comment by NisugaJ** [comment]
> @Nykollas @NisugaJ thanks for raising it. Can you guys try `v3.2.2` and check if it is still happening? I've done a couple fixes on the Autoplay regarding having an empty children state.
> 
> @NisugaJ Either way you could add a startupScreen component or avoid the slider to be mounted when there's no `data` on the `allPhotos` state. That could also do it.

@rcaferati  thank you for suggesting but it is **not working**. I upgraded to 3.2.2 and 
added a startupScreen too. 
Added some separate images to the children too.
Seems like AutoPlay is not working properly with startUpScreen. 

I think startUpScreen is not waiting until all data is loaded. It waits only for first child.
at least  loremPics in the middle didnot  display. 

However, I am pretty sure that startUpScreen is not waiting until all children are settled. It must wait for all the children. Otherwise autoplay gets paused.

**And startUpScreen must have  a very strict connection with data-fetching.**
**these are my suggestioins. Thank you. Hope you will be making it perfect for a perfect data-fetch : ) **
![image](https://user-images.githubusercontent.com/44756803/73370651-921e8980-42da-11ea-8987-b2c453a0d25c.png)

It gives me 
TypeError: Cannot read property 'map' of null  as I didn't wait for data-fetch
